src/effect.ts

```diff
@@ -79,7 +79,7 @@ export class Effect {
   }
 }
 
-export function createEffect(fn: () => void, deps?: Array<ReactiveValue<unknown>>): () => boolean {
+export function createEffect<T>(fn: () => void, deps?: Array<ReactiveValue<T>>): () => boolean {
```

Comment:
This is the first step toward solving this; there is *one spot* which doesn't like it, but it's one that I think is going to be annoying to propagate elsewhere.